src/action/base/setup_default_profile.rs

```diff
@@ -104,9 +115,18 @@ impl Action for SetupDefaultProfile {
             )));
         };
 
-        {
-            let reginfo_path =
-                Path::new(crate::action::base::move_unpacked_nix::DEST).join(".reginfo");
+        let found_nix_paths = glob::glob(&format!("{}/nix-*", unpacked_path.display()))
+            .map_err(|e| ActionError::Custom(Box::new(e)))?
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| ActionError::Custom(Box::new(e)))?;
+        assert_eq!(
+            found_nix_paths.len(),
+            1,
+            "Did not expect to find multiple nix paths, please report this"
+        );
+        let found_nix_path = found_nix_paths.into_iter().next().unwrap();
+        let reginfo_path = PathBuf::from(found_nix_path).join(".reginfo");
+        if reginfo_path.exists() {
```

Comment:
I think if there's no reginfo, that should be an error; otherwise, Nix's db will not contain any of the paths from this tarball and will delete them on the next GC, I believe (untested though, could be very wrong).